CMakeLists.txt

```diff
@@ -1,48 +1,51 @@
-cmake_minimum_required(VERSION 3.7...3.29)
-
-if (${CMAKE_VERSION} VERSION_LESS 3.12)
-    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
-endif ()
+cmake_minimum_required(VERSION 3.23...3.29)
 
 project(WaraPSClient VERSION 1.0
         DESCRIPTION "WaraPS Client"
         LANGUAGES CXX
 )
 
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Consider setting these three values before the first `project` command. Also think about if others include this CMakeLists.txt file as part of a bigger build and you thus should check if e.g. the standard is set to at least C++20 otherwise you abort,  and if not set you accept the possibly higher value.